Source/CarthageKit/Cartfile.swift

```diff
@@ -92,6 +92,27 @@ extension Cartfile: Printable {
 	}
 }
 
+// Duplicate dependencies
+extension Cartfile {
+	/// Returns an array containing projects that are listed as duplicate
+	/// dependencies.
+	public func duplicateProjects() -> [ProjectIdentifier] {
+		let projectSet = buildCountedSet(self.dependencies.map { $0.project })
+		return filter(projectSet) { $0.1 > 1}
+			.map { $0.0 }
+	}
+}
+
+/// Returns an array containing projects that are listed as dependencies
+/// in both arguments.
+public func duplicateProjectsInCartfiles(cartfile1: Cartfile, cartfile2: Cartfile) -> [ProjectIdentifier] {
+	let projectSet1 = buildCountedSet(cartfile1.dependencies.map { $0.project }).keys
```

Comment:
By immediately converting from a dictionary to an ordered collection, we lose O(1) lookups. Can this wait until after the `filter`?